subsys/ipc/ipc_service/backends/CMakeLists.txt

```diff
@@ -5,3 +5,6 @@ zephyr_sources_ifdef(CONFIG_IPC_SERVICE_BACKEND_ICMSG_ME_INITIATOR	ipc_icmsg_me_
 zephyr_sources_ifdef(CONFIG_IPC_SERVICE_BACKEND_ICMSG_ME_FOLLOWER	ipc_icmsg_me_follower.c)
 zephyr_sources_ifdef(CONFIG_IPC_SERVICE_BACKEND_ICBMSG			ipc_icbmsg.c)
 zephyr_sources_ifdef(CONFIG_IPC_SERVICE_BACKEND_RPMSG			ipc_rpmsg_static_vrings.c)
+
+# For strnlen()
+zephyr_compile_definitions_ifdef(CONFIG_IPC_SERVICE_BACKEND_ICBMSG _POSIX_C_SOURCE=200809L)
```

Comment:
zephyr_compile_definitions_ifdef() adds that option to the whole zephyr and somebody may complain if they tried to define it to something else and get a warning.
you can just add it locally for the file where you need it.
I'd recommend just doing this:
```diff
diff --git a/subsys/ipc/ipc_service/backends/ipc_icbmsg.c b/subsys/ipc/ipc_service/backends/ipc_icbmsg.c
index da2db4ccbc8..c7cb5f8ea34 100644
--- a/subsys/ipc/ipc_service/backends/ipc_icbmsg.c
+++ b/subsys/ipc/ipc_service/backends/ipc_icbmsg.c
@@ -75,6 +75,9 @@
  * data messages, it calls bound endpoint and it is ready to send data.
  */
+#undef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L /* For strnlen() */
+
#include <string.h>
#include <zephyr/logging/log.h>
```

Reply:
I agree. Done